Development/Porting.md

```diff
@@ -5,86 +5,220 @@ permalink: /Porting/
+4. External programmer: an external programmer is required to flash Heads and, if necessary, to recover from a brick.
+5. Users and Board-testers: There should be multiple users and testers interested in using the board, as this makes maintenance and testing easier and benefits the entire ecosystem. You may, of course, port it for yourself first and allow others to join later. However, you must commit to testing—especially after coreboot version bumps or Linux kernel updates. If testing is not done in a timely manner, the board will be moved to an "unmaintained, untested" status. This would be unfortunate, potentially a waste of time, and disappointing for everyone involved.
+6. GPU: dGPUs are problematic in Heads for various reasons. While successful ports on older machines with dGPUs exist (TODO: Add link), security may be affected, and there is currently no clear solution for that. (TODO: Link to Matrix coreboot discussion)
 
-* Either copy one of the existing config files, such as the
-`config/coreboot-x230.config` to `config/coreboot-newarch.conf`,
-or use your existing coreboot `.config` file in its place.  You'll want to
-be sure that it is using a Linux payload named `./bzImage` and a Linux initrd
-named `./initrd.cpio.xz`.  No command line options are necessary.
+The scheme depicts a port cycle. 
+![Port cycle]({{ site.baseurl }}/images/Porting_cycle.png)
+TODO (remove source code after feedback)
+```mermaid
+  graph TD;
+      build_rom-->flash
+      flash-->test;
+      test-->debugg;
+      debugg-->test;
+      debugg-->remove_debugging_mode;
+      remove_debugging_mode-->write_flashing/dissassambly_guide_on_heads-wiki;
+      write_flashing/dissassambly_guide_on_heads-wiki-->maintance
+```
+Files needed to be created/modified are depicted on the scheme.
+![Files overview]({{ site.baseurl }}/images/Files_scheme.png)
+
+TODO (remove source code after feedback)
+```mermaid
+  graph LR;
+      heads/-->config/;
+      config/-->coreboot-new_board.config;
+      config/-->linux-new_board.config;
+      heads/-->boards/new_board-hotp-maximized/;
+      heads/-->boards/new_board-maximized/;
+      boards/new_board-hotp-maximized/-->new_board-hotp-maximized.config;
+      boards/new_board-maximized/-->new_board-maximized.config;
+      heads/-->blobs/new_board/;
+      blobs/new_board/-->ME.bin;
+      blobs/new_board/-->IFD.bin;
+      blobs/new_board/-->GBE.bin;
+      heads/-->targets/;
+      targets/-->new_board.mk
+      heads/-->modules/;
+      modules/-->coreboot
+      heads/-->.circleci/;
+      .circleci/-->config.yml
+```
+
+Prerequisites for building the ROM
+===
+The goal is to build a ROM before the code review. `new_board` refers to the name of your board, e.g., `t480`
+
+coreboot.config:
```

Comment:
Not sure how to add IFD <-> coreboot CBFS region <-> ME neutering freed space transfer here.


Maybe add a TODO where you see fit.

Rationale:
- If ME neutered, ME region defined under IFD must fit.
- With IFD ME region reduced, BIOS region can grow with ME freed space
- With BIOS region augmented, CBFS region of coreboot config must be augmented to fit "maximized" space 

Ref: https://osresearch.net/Prerequisites#legacy-vs-maximized-boards